modules/xmpp/strophe.ping.js

```diff
@@ -141,6 +157,52 @@ class PingConnectionPlugin extends ConnectionPlugin {
             logger.info('Ping interval cleared');
         }
     }
+
+    /**
+     * Adds the current time to the array of send ping timestamps.
+     * @private
+     */
+    _addPingExecutionTimestamp() {
+        this.pingExecIntervals.push(new Date().getTime());
+
+        // keep array length to PING_TIMESTAMPS_TO_KEEP
+        if (this.pingExecIntervals.length > PING_TIMESTAMPS_TO_KEEP) {
+            this.pingExecIntervals.shift();
+        }
+    }
+
+    /**
+     * Was ping suspended
+     * Checks the maximum gap between sending pings, considering and the
+     * current time. Trying to detect computer inactivity (sleep).
+     *
+     * @returns {boolean} whether sending pings was suspended for a long period
+     * of time.
+     */
+    wasPingSuspended() {
+        const pingIntervals = this.pingExecIntervals.slice();
+
+        // we need current time, as if ping was sent now
+        // if computer sleeps we will get correct interval after next
+        // scheduled ping, bet we sometimes need that interval before waiting
+        // for the next ping, on closing the connection on error.
+        pingIntervals.push(new Date().getTime());
```

Comment:
Do you intentionally not use _addPingExecutionTimestamp here? It looks like calls to wasPingSuspended can make the array grow (and _addPingExecutionTimestamp will not shrink it back, as it only removes 1 element).

Reply:
Well my thinking was working on a copy of the array and compute it as if a ping was just sent it, does it make sense. It is a copy of the array, so nothing grows.